lib/fog/openstack/core.rb

```diff
@@ -145,7 +145,7 @@ def microversion_newer_than?(version)
 
       def headers(additional_headers)
         additional_headers ||= {}
-        if @microversion
+        unless @microversion.nil? or @microversion.empty?
```

Comment:
Please use `&&` operator instead of `or`.

Reply:
@gildub,  Don't you mean `||` instead of `or`?

Reply:
Yes indeed! `or` as well as `and` are not recommended.